dask/array/core.py

```diff
@@ -2568,32 +2568,83 @@ def concatenate(seq, axis=0, allow_unknown_chunksizes=False):
     return Array(dsk2, name, chunks, dtype=dt)
 
 
-def insert_to_ooc(arr, out, lock=True, region=None):
-    if lock is True:
-        lock = Lock()
+def insert_to_ooc_store(x, out, index, lock, region):
```

Comment:
This function name will end up in diagnostics.  We probably want it to remain short, like `store`.

Reply:
or `_store`

Reply:
Otherwise everything here seems fine to me

Reply:
Renamed to `ooc_store`. Would that work?

Reply:
We understand `ooc`, but users may not.  Also, this function is now being used in distributed contexts, so "out of core" may not be the best name for any of these functions.

Reply:
No strong objection, although it's not clear to me what `sel_` stands for.

Although actually, I should revise my previous statement.  This isn't the name that will show up, it's the keyname.  I care far more about that than this function name.  I apologize for the mixup.

Reply:
Was meaning selection, but yeah I agree. Was thinking about renaming to something like `store_chunk`. Still short, but more meaningful.

Ah ok. Though was pretty lazy with the function name. So having a better one is a good idea.

Reply:
Fixed to `store_chunk`